tests/test_custom_type_setup.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+
+import gc
+import weakref
+
+import pytest
+
+import env  # noqa: F401
+from pybind11_tests import custom_type_setup as m
+
+
+@pytest.fixture
+def gc_tester():
+    """Tests that an object is garbage collected."""
```

Comment:
Might be helpful to add something like:

Assumes that the Python VM including garbage collector are designed to satisfy this test. This is generally true for CPython, but is known to not be true for PyPy.

Reply:
Done.